core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -825,6 +840,50 @@ public CacheEntry<K, V> fromCacheEntry(CacheEntry entry) {
       }
    }
 
+   private class SizeComposedType implements ComposedType<K, Long, Long> {
+
+      @Override
+      public CompletionStage<PublisherResult<Long>> localInvocation(boolean parallelPublisher,
+         IntSet segments, Set<K> keysToInclude, Set<K> keysToExclude, long explicitFlags,
+         DeliveryGuarantee deliveryGuarantee,
+         Function<? super Publisher<Long>, ? extends CompletionStage<Long>> transformer,
+         Function<? super Publisher<Long>, ? extends CompletionStage<Long>> finalizer) {
+         return KEY_COMPOSED.localInvocation(parallelPublisher, segments, keysToInclude, keysToExclude,
+               explicitFlags, deliveryGuarantee, transformer, finalizer);
+      }
+
+      @Override
+      public SizeCommand remoteInvocation(boolean parallelPublisher, IntSet segments,
+                                          Set<K> keysToInclude, Set<K> keysToExclude,
+                                          long explicitFlags,
+                                          DeliveryGuarantee deliveryGuarantee,
+                                          Function<? super Publisher<Long>, ? extends CompletionStage<Long>> transformer,
+                                          Function<? super Publisher<Long>, ? extends CompletionStage<Long>> finalizer) {
+         return commandsFactory.buildSizeCommand(segments, keysToExclude, explicitFlags);
+      }
+
+      @Override
+      public CompletionStage<PublisherResult<Long>> contextInvocation(IntSet segments, Set<K> keysToInclude,
+         InvocationContext ctx, Function<? super Publisher<Long>, ? extends CompletionStage<Long>> transformer) {
+         return KEY_COMPOSED.contextInvocation(segments, keysToInclude, ctx, transformer);
```

Comment:
This shouldn't ever be used. If the user provides a context that is not empty we either have to skip the optimization or we could do the optimization for segments that aren't in the context. I would say for this PR just always disable the optimization if the context has any values.